openfecli/commands/gather_septop.py

```diff
@@ -131,16 +132,16 @@ def _get_names(result: dict) -> tuple[str, str]:
     tuple[str, str]
         Ligand names corresponding to the results.
     """
-    # TODO: use gather's _get_names once it is improves (uses input.stateA/B.name)
-    try:
-        nm = list(result["unit_results"].values())[0]["name"]
 
-    except KeyError:
-        raise ValueError("Failed to guess names")
+    # TODO: do we always want to treat "solvent" as the source of truth? should we verify that "complex" has the same name?
+    # TODO: is it faster to only load the ligands, or is loading the entire pur cheap?
+    solvent_data = list(result["protocol_result"]["data"]["solvent"].values())[0][0]
+    pur_solvent = gufe.ProtocolUnitResult.from_dict(solvent_data)
+
+    name_A = pur_solvent.inputs["alchemical_components"]["stateA"][0].name
+    name_B = pur_solvent.inputs["alchemical_components"]["stateB"][0].name
```

Comment:
soliciting @hannahbaumann's feedback on if this is the most robust way to pull names from septop protocol unit results